innovation.scss

```diff
@@ -0,0 +1,4482 @@
+		}
+		&.hexagon_icon_199 {
+			background-position: -412px -1826px;
+		}
+		&.hexagon_icon_200 {
+			background-position: -513px -1826px;
+		}
+		&.hexagon_icon_201 {
+			background-position: -614px -1826px;
+		}
+		&.hexagon_icon_202 {
+			background-position: -715px -1826px;
+		}
+		&.hexagon_icon_203 {
+			background-position: -816px -1826px;
+		}
+		&.hexagon_icon_204 {
+			background-position: -917px -1826px;
+		}
+		&.hexagon_icon_205 {
+			background-position: -8px -1927px;
+		}
+		&.hexagon_icon_206 {
+			background-position: -109px -1927px;
+		}
+		&.hexagon_icon_207 {
+			background-position: -210px -1927px;
+		}
+		&.hexagon_icon_208 {
+			background-position: -311px -1927px;
+		}
+		&.hexagon_icon_209 {
+			background-position: -412px -1927px;
+		}
+		&.hexagon_icon_210 {
+			background-position: -513px -1927px;
+		}
+		&.hexagon_icon_211 {
+			background-position: -614px -1927px;
+		}
+		&.hexagon_icon_212 {
+			background-position: -715px -1927px;
+		}
+		&.hexagon_icon_213 {
+			background-position: -816px -1927px;
+		}
+		&.hexagon_icon_214 {
+			background-position: -917px -1927px;
+		}
+		&.hexagon_icon_215 {
+			background-position: -1018px -1119px;
+		}
+		&.hexagon_icon_216 {
+			background-position: -1119px -1119px;
+		}
+		&.hexagon_icon_217 {
+			background-position: -1220px -1119px;
+		}
+		&.hexagon_icon_218 {
+			background-position: -1321px -1119px;
+		}
+		&.hexagon_icon_219 {
+			background-position: -1422px -1119px;
+		}
+	}
+} 
+
+/* scss_generator code starts here */
+/* anything below this will get blown away on rebuild */
+
+.font_size_1 {
+	font-size: 0.0625rem;
+}
+
+.font_size_2 {
+	font-size: 0.125rem;
+}
+
+.font_size_3 {
+	font-size: 0.1875rem;
+}
+
+.font_size_4 {
+	font-size: 0.25rem;
+}
+
+.font_size_5 {
+	font-size: 0.3125rem;
+}
+
+.font_size_6 {
+	font-size: 0.375rem;
+}
+
+.font_size_7 {
+	font-size: 0.4375rem;
+}
+
+.font_size_8 {
+	font-size: 0.5rem;
+}
+
+.font_size_9 {
+	font-size: 0.5625rem;
+}
+
+.font_size_10 {
+	font-size: 0.625rem;
+}
+
+.font_size_11 {
+	font-size: 0.6875rem;
+}
+
+.font_size_12 {
+	font-size: 0.75rem;
+}
+
+.font_size_13 {
+	font-size: 0.8125rem;
+}
+
+.font_size_14 {
+	font-size: 0.875rem;
+}
+
+.font_size_15 {
+	font-size: 0.9375rem;
+}
+
+.font_size_16 {
+	font-size: 1.0rem;
+}
+
+.font_size_17 {
+	font-size: 1.0625rem;
+}
+
+.font_size_18 {
+	font-size: 1.125rem;
+}
+
+.font_size_19 {
+	font-size: 1.1875rem;
+}
+
+.font_size_20 {
+	font-size: 1.25rem;
+}
+
+.font_size_21 {
+	font-size: 1.3125rem;
+}
+
+.font_size_22 {
+	font-size: 1.375rem;
+}
+
+.font_size_23 {
+	font-size: 1.4375rem;
+}
+
+.font_size_24 {
+	font-size: 1.5rem;
+}
+
+.font_size_25 {
+	font-size: 1.5625rem;
+}
+
+.font_size_26 {
+	font-size: 1.625rem;
+}
+
+.font_size_27 {
+	font-size: 1.6875rem;
+}
+
+.font_size_28 {
+	font-size: 1.75rem;
+}
+
+.font_size_29 {
+	font-size: 1.8125rem;
+}
+
+.font_size_30 {
+	font-size: 1.875rem;
+}
+
+.M { 
+	&.hexagon_card_icon {
```

Comment:
Why is hexagon_icon_0 defined 4 times? I think we have all of the hex icons defined twice (which is thousands of lines of code)

Reply:
It was an issue with me not manually deleting a section of code and then I changed the string on the css generator and so it added it's output to the end of the old stuff that did not get deleted.

So the correct number is two and not 4. Though getting it down to one would be cool and at the top of my head if I generated everything in a json format and then wrote the json to scss it could refactor that down. Not sure how much of a benefit that would be right now

Reply:
Adding another level on top of this seems like overkill!

Reply:
It would be run the loops and build a large json object -> feed that json object into a scss printer. so instead of two sizes of .[size]&.[class1]&.[class2] { code } it would be .[class1]&.[class2] { &[size1] {code} &.[size2] {code} }